fe/src/main/java/org/apache/doris/load/loadv2/LoadJob.java

```diff
@@ -303,18 +317,54 @@ public void cancelJobWithoutCheck(FailMsg failMsg) {
     public void cancelJob(FailMsg failMsg) throws DdlException {
         writeLock();
         try {
+            // check
             if (isCommitting) {
                 LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
                                  .add("error_msg", "The txn which belongs to job is committing. "
                                          + "The job could not be cancelled in this step").build());
                 throw new DdlException("Job could not be cancelled while txn is committing");
             }
+            if (isFinished()) {
+                LOG.warn(new LogBuilder(LogKey.LOAD_JOB, id)
+                                 .add("state", state)
+                                 .add("error_msg", "Job could not be cancelled when job is finished or cancelled")
+                                 .build());
+                throw new DdlException("Job could not be cancelled when job is finished or cancelled");
+            }
+
+            checkAuth();
             executeCancel(failMsg);
         } finally {
             writeUnlock();
         }
     }
 
+    public void checkAuth() throws DdlException {
```

Comment:
It is not a good idea that use `tableNames` to infer whether it is a `CANCEL` command

you can change the function to 

```suggestion
    public void checkAuth(Set<String> tableNames) throws DdlException {
```

Reply:
Whether it is a CANCEL command, the tableNames is the required param of `checkAuth`. So maybe getting it inside of function is better?

Reply:
It is better that `checkAuth` is a private method. If this is a public function, some caller may forget call this function before call other method